Client/Frontend/Browser/URLBarView.swift

```diff
@@ -620,8 +620,8 @@ extension URLBarView: TabLocationViewDelegate {
 
     func tabLocationViewDidTapLocation(tabLocationView: TabLocationView) {
         var locationText = delegate?.urlBarDisplayTextForURL(locationView.url)
-        if let host = locationView.url?.host {
-            locationText = locationView.url?.absoluteString?.stringByReplacingOccurrencesOfString(host, withString: host.asciiHostToUTF8())
+        if let text = locationText, let url = NSURL(string: text), let host = url.host {
```

Comment:
Apologies. Combining a fix for .ca TLD's and the actual bug is making things confusing. 

This change here is the actual bug. `locationView.url.host` will exist every time a valid URL is used in the url bar. So that line to check for a host `locationView.url?.host` is clobbering whatever is returned from `urlBarDisplayTextForURL`.

`urlBarDisplayTextForURL` is responsible for parsing out the search term. But what this function does is either return the search term OR the full url. So it's better to compare that than getting `locationView.url`